core/domain/wipeout_service.py

```diff
@@ -1253,3 +1255,88 @@ def _pseudonymize_models_transactional(voiceover_application_models):
             voiceover_application_models[
                 i:i + feconf.MAX_NUMBER_OF_OPS_IN_TRANSACTION]
         )
+
+
+def _pseudonymize_blog_posts_models(pending_deletion_request):
+    """Pseudonymizes the blog post models for the user with user_id.
+
+    Args:
+        pending_deletion_request: PendingDeletionRequest. The pending
+            deletion request object to be saved in the datastore.
+    """
+    user_id = pending_deletion_request.user_id
+
+    # We want to preserve the same pseudonymous user ID on all the models
+    # related to one blog post. So we collect all the users' blog
+    # post models and blog post summary models then
+    # we generate a pseudonymous user ID and replace the user ID
+    # with that pseudonymous user ID in all the models.
+    blog_post_model_class = blog_post_models.BlogPostModel
+    blog_post_models_list = blog_post_model_class.query(
+        datastore_services.any_of(
+            blog_post_model_class.author_id == user_id)
```

Comment:
Redundant, please remove:
```suggestion
        blog_post_model_class.author_id == user_id
```

Reply:
Removed.